listComprehensions.py

```diff
@@ -0,0 +1,37 @@
+"""
+CP1404/CP5632 Practical
+List comprehensions
+"""
+
+def main():
+    names = ["Bob", "Angel", "Jimi", "Alan", "Ada"]
+    full_names = ["Bob Martin", "Angel Harlem", "Jimi Hendrix", "Alan Turing", "Ada Lovelace"]
+
+    # for loop that creates a new list containing the first letter of each name
+    first_initials = []
+    for name in names:
+        first_initials.append(name[0])
+        print(first_initials)
+
+    # list comprehension that does the same thing
+    first_initials = [name[0] for name in names]
+    print(first_initials)
+
+    # list comprehension that creates a list containing tuples of both initials
+    # splits each name and adds the first letter of each part to a tuple
```

Comment:
As it is a quiet the chunk of commenting might be worth putting it in to a docstring.